libnavigation-metrics/src/test/java/com/mapbox/navigation/metrics/MainCoroutineRule.kt

```diff
@@ -0,0 +1,33 @@
+package com.mapbox.navigation.metrics
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runBlockingTest
+import kotlinx.coroutines.test.setMain
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+@ExperimentalCoroutinesApi
+class MainCoroutineRule(
```

Comment:
This should be reused - https://github.com/mapbox/mapbox-navigation-android/issues/2411. Not blocking the PR though.

Reply:
Yeah I'm agree
But, let's first merge this PR, and in separate PR we will transform `libtesting-ui` to generic module and use `MainCoroutineRule` across all SDK from there. wdyt?

Reply:
Sounds good :+1: